filters/auth/grantconfig.go

```diff
@@ -371,7 +371,9 @@ func (c *OAuthConfig) RedirectURLs(req *http.Request) (redirect, original string
 	original = u.String()
 
 	u.Path = c.CallbackPath
+	u.RawPath = ""
```

Comment:
`RawPath` is a hint how to encode `Path` (see https://pkg.go.dev/net/url#URL.EscapedPath), since we change `Path`, reset `RawPath` as well.

Reply:
maybe we should have these comments as comments in code?

Reply:
I don't like that we reused original url object to construct redirect url so I rewrote it to be constructed from scratch. This was we don't need to reset any fields.